AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -249,7 +268,7 @@ - (void)startUpdate {
       return;
     }
     NSString *distributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
-    if (updateToken || distributionGroupId) {
+    if (updateToken || distributionGroupId || MSDistribute.updateTrack == MSUpdateTrackPublic) {
       [self checkLatestRelease:updateToken distributionGroupId:distributionGroupId releaseHash:releaseHash];
```

Comment:
If switch from private to public, this won't work as this code will pass token and private url while track is public. Either pass nil when public (even if we have token), or pass track to the method.